### PR TITLE
Gruntfile.js: fix for require exports

### PR DESCRIPTION
**Pull request by Badalik**

### CODE REVIEW

**Comment by staabm** [comment]
@Badalik whats the background of the change? Why is it required, what is the issue?

I dont have experience with grunt so, I cant merge it without enough information from your side.

@Arkni maybe you can have a 2nd look at it.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
-dist
+#dist
```
Comment (Arkni) [inline .gitignore]: Please revert this change.

**Comment by Arkni** [comment]
You should not include the dist folder (and its files), please remove them and then push again.

File: Gruntfile.js [hunk 2]
```diff
@@ -25,7 +25,9 @@ banner = "/*!\n" +
 umdStart = "(function( factory ) {\n" +
 	"\tif ( typeof define === \"function\" && define.amd ) {\n";
 
-umdMiddle = "\t} else {\n" +
+umdMiddle = "\t} else if (typeof exports === \"object\") {\n" +
```
Comment (Arkni) [inline Gruntfile.js]: I think you want to add CommonJS pattern. If that so, you have to change this line to:
```js
} else if ( typeof module === 'object' && module.exports ) {
```

File: Gruntfile.js [hunk 3]
```diff
@@ -25,7 +25,9 @@ banner = "/*!\n" +
...
+	"\t\tfactory(require(\"jquery\"));\n" +
```
Comment (Arkni) [inline Gruntfile.js]: This line should be changed to:
```js
module.exports = factory( require( "jquery" ) );
```

**Comment by Badalik** [comment]
I work with a template on which EC6 then use babel.js transforms in EU5. We use import EC6 and npm require and without the code plugins are not working.

**Comment by Badalik** [comment]
You can make it in his own way, I just have a problem. This is a useful editing

**Comment by Arkni** [comment]
@Badalik 
Did you try to test the suggested changes ?

**Comment by staabm** [comment]
as far as I can see we already have landed a similar path with PR #1621.

please shout out if I miss something.
